Lab1/app/src/main/java/com/mobileproto/jovanduy/lab1/MainActivity.java

```diff
@@ -0,0 +1,234 @@
+                itemsAdapter.notifyDataSetChanged();
+            }
+        });
+
+        // do nothing when negative button is clicked
+        deleteBuilder.setNegativeButton("Cancel", new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                return;
+            }
+        });
+
+        return deleteBuilder.create();
+    }
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        items = new ArrayList<Item>();
+        itemsAdapter = new NewAdapter(this, items);
+
+        listView = (ListView) findViewById(R.id.lvItems);
+        listView.setAdapter(itemsAdapter);
+        // edit item when clicked
+        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                openEditor(position).show();
+            }
+        });
+        // delete item when long clicked
+        listView.setOnItemLongClickListener(new AdapterView.OnItemLongClickListener() {
+            @Override
+            public boolean onItemLongClick(AdapterView<?> parent, View view, int position, long id) {
+                openDeleter(position).show();
+                return true;
+            }
+        });
+
+        // button for adding new items
+        addButton = (Button) findViewById(R.id.addButton);
+        // open AlertDialog to add item when button is clicked
+        addButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                openAdder().show();
+            }
+        });
+
+        // button for deleting done items
+        deleteButton = (Button) findViewById(R.id.deleteButton);
+        // open AlertDialog to delete done items when button is clicked
+        deleteButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                openDoneDeleter().show();
+            }
+        });
+
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        // Inflate the menu; this adds items to the action bar if it is present.
+        getMenuInflater().inflate(R.menu.menu_main, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        // Handle action bar item clicks here. The action bar will
+        // automatically handle clicks on the Home/Up button, so long
+        // as you specify a parent activity in AndroidManifest.xml.
+        int id = item.getItemId();
+
+        //noinspection SimplifiableIfStatement
+        if (id == R.id.action_settings) {
+            return true;
+        }
+
+        return super.onOptionsItemSelected(item);
+    }
+}
```

Comment:
Nice job separating stuff out into various methods. Generally, you can just comment once at the top of the method to say what it does, and get rid of comments throughout.